File: factors.c
#include <stdio.h>

void factorize(int n)
{
	int factor = 2;

	printf("%d=", n);

	while (n > 1)
	{
		if (n % factor == 0)
		{
			printf("%d", factor);
			n /= factor;

			if (n > 1)
			{
				printf("*");
			}
		}
		else
		{
			factor++;
		}
	}

	printf("\n");
}

int main(int argc, char *argv[])
{
	FILE *file;
	int number;

	if (argc != 2)
	{
		printf("Usage: %s <file>\n", argv[0]);
		return (1);
	}

	file = fopen(argv[1], "r");

	if (file == NULL)
	{
		printf("Error opening the file.\n");
		return (1);
	}

	while (fscanf(file, "%d", &number) == 1)
	{
		factorize(number);
	}

	fclose(file);

	return (0);
}

File: factors.py
#!/usr/bin/python3
import sys


def factorize(number):
    # Start with the smallest possible factor
    factor = 2
    while factor * factor <= number:
        if number % factor == 0:
            # Found a factor, print the factorization
            other_factor = number // factor
            print(f"{number}={factor}*{other_factor}")
            return
        factor += 1

    # If the number is a prime, print it as a factorization of itself
    print(f"{number}={number}*1")


def main(filename):
    with open(filename, 'r') as file:
        for line in file:
            # Convert each line to an integer and factorize
            number = int(line.strip())
            factorize(number)


if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: factors <file>")
        sys.exit(1)

    filename = sys.argv[1]
    main(filename)
